lib/Data/Chronicle/Reader.pm

```diff
@@ -130,7 +130,7 @@ sub get {
 
     if (blessed($self->cache_reader)) {
         my $cached_data = $self->cache_reader->get($key);
-        return JSON::from_json($cached_data) if defined $cached_data;
+        return JSON::MaybeXS->new->decode(decode_utf8($cached_data)) if defined $cached_data;
```

Comment:
Are you sure, that we need do  utf8-decoding here? We don't do that for DB-value. Also we don't do reverse operation (`encode`) in `set`, when we put value in redis

Reply:
There is `encode_utf8` - line 151 of Writer.pm. And here we're working only from Redis, leaving DB part intact.
And this encode/decode is only for converting between UTF-8 and Perl's representation of data, AFAIK

Reply:
@bork1n I cannot find `encode_utf8` at https://github.com/bork1n/perl-Data-Chronicle/blob/02ac2e46def431e97971002bb2b185fa7fbd9141/lib/Data/Chronicle/Writer.pm#L151 . 

Could you point, please?

Reply:
https://github.com/binary-com/perl-Data-Chronicle/pull/24/files#diff-41cc188769b5fc3d9e851332c19c7556L151